minimal-plugin/src/main/java/io/micronaut/gradle/MicronautMinimalApplicationPlugin.java

```diff
@@ -212,6 +216,10 @@ private void configureGoogleCloudFunctionRuntime(Project project, Project p, Dep
         });
 
         // Google Cloud Function requires shadow packaging
-        p.getPluginManager().apply(ShadowPluginSupport.SHADOW_PLUGIN);
+        try {
+            p.getPluginManager().apply(ShadowPluginSupport.SHADOW_PLUGIN);
```

Comment:
:x: Why this try catch? Looks rather nasty. If we have to do this, then something needs to be fixed.

Reply:
It is necessary because I GCP requires the shadowJAR plugin. 


These entries in the `where` table: 
```
 'google_function'      | ...
 'google_function'      | ...
 'google_function'      | ...
```

cause that exception to be trigger. 

However, I don't think I can apply shadow Jar gradle plugin in this test. 

I think I could if move he test to `functional-test`

Reply:
Right, so yes, if the plugin requires 3rd party the test should be moved to functional tests project. It's a bit cumbersome but it's actually the point of separating out: it allows catching errors which weren't found before (dependencies accidentally on classpath)

Reply:
I have moved to functional